ui/app/components/mount-backend/type-form.js

```diff
@@ -26,11 +25,15 @@ export default class MountBackendTypeForm extends Component {
   @service version;
 
   get secretEngines() {
-    return this.version.isEnterprise ? allEngines() : mountableEngines();
+    return this.version.isEnterprise
+      ? ALL_ENGINES.filter((engine) => engine.mountType !== 'auth')
```

Comment:
What about engines that support both secret and auth types?

Currently, it's not intuitive that “both” is a valid option, and I suspect this could cause confusion in other parts of the code as well.

I suggest updating the mountType in `all-engines-metadata.js` to be an array instead of a single string. For example:
```
mountType: ['secret', 'auth']
```
Then, wherever we need to check a type, we can use .includes() instead of checking for an exact string.

What do you think about this approach?